test/integration/segmentation_camera_plugin.cc

```diff
@@ -47,43 +47,46 @@
 #include "TransportTestTools.hh"
 
 using namespace ignition;
-using namespace rendering;
-
-class SegmentationCameraSensorTest: public testing::Test,
-  public testing::WithParamInterface<const char *>
-{
-  // Create a Segmentation Camera sensor from a SDF and gets a image message
-  public: void ImagesWithBuiltinSDF(const std::string &_renderEngine);
-};
 
 /// \brief mutex for thread safety
 std::mutex g_mutex;
 
 /// \brief Segmentation buffer
-unsigned char *g_buffer = nullptr;
+uint8_t *g_buffer = nullptr;
 
 /// \brief counter of received segmentation msgs
 int g_counter = 0;
 
+/// \brief Label of the hidden box
+uint8_t hiddenLabel = 4;
+
 /// \brief callback to get the segmentation buffer
 void OnNewSegmentationFrame(const msgs::Image &_msg)
 {
   g_mutex.lock();
+
+  // the image has 3 channels
   unsigned int size = _msg.width() * _msg.height() * 3;
 
   if (!g_buffer)
-    g_buffer = new unsigned char[size];
+    g_buffer = new uint8_t[size];
 
   memcpy(g_buffer, _msg.data().c_str(), size);
   g_counter++;
   g_mutex.unlock();
 }
 
 /// \brief wait till you read the published frame
-void WaitForNewFrame()
+void WaitForNewFrame(ignition::sensors::Manager &mgr)
 {
+  // wait for a few segmentation camera frames
+  mgr.RunOnce(std::chrono::steady_clock::duration::zero(), true);
+
+  g_counter = 0;
```

Comment:
```suggestion
  g_counter = 0;
  
  // wait for a few segmentation camera frames
  mgr.RunOnce(std::chrono::steady_clock::duration::zero(), true);
```
In order to prevent any weird timing bugs (especially on server machines where computing speed varies), I believe the counter should be set to zero before `RunOnce`. That way, we don't run the risk of receiving the image before setting the counter to 0 and then overriding that by setting the counter to 0 before we check if something was received in the loop below